addons/mail/static/tests/message/message.test.js

```diff
@@ -1964,3 +1964,17 @@ test("Copy Message Link", async () => {
     await click(".o-mail-Composer-send:enabled");
     await contains(".o-mail-Message", { text: url(`/mail/message/${messageId_2}`) });
 });
+
+test("should remove link preview when message no longer contains the link", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "EditPreviewTest" });
+    await start();
+    await openDiscuss(channelId);
+    await insertText(".o-mail-Composer-input", "https://make-link-preview.com");
+    await click(".o-mail-Composer-send:enabled");
+    await click(".o-mail-Message [title='Expand']");
+    await click(".o-mail-Message-moreMenu [title='Edit']");
+    await insertText(".o-mail-Message.o-editing .o-mail-Composer-input", "Hi", { replace: true });
+    await press("Enter");
+    await contains(".o-mail-LinkPreviewCard", { count: 0 });
```

Comment:
There needs a positive test for the element before doing a negative. So, the test case doesn't pass in case there has been some changes in the loading of the element or the selector for the class.